edx_solutions_api_integration/users/views.py

```diff
@@ -1413,51 +1413,17 @@ def get(self, request, *args, **kwargs):  # pylint: disable=unused-arguments
             cache_course_user_data('social', course_id, user.id, {'score': social_engagement_score})
         else:
             data = cached_social_data
+
         if include_stats:
-            data['stats'] = self._get_user_discussion_metrics(request, user, course_id)
+            if not data.get('stats'):
+                data['stats'] = StudentSocialEngagementScore.get_user_engagements_stats(course_key, user.id)
+                cache_course_user_data('social', course_id, user.id, {'score': data['score'], 'stats': data['stats']})
```

Comment:
This ends up throwing away the `course_avg` I think. The next time we run this for a user, `cached_social_data` will exist but without `course_avg` and it'll just move on.

Reply:
I've tested it and after doing `cache.clear()` and `course_avg` is still in the right place. I wasn't digging into it to see why it works, but my quick guess is that `cache_course_data` and `cache_course_user_data` save it differently (for the first the `course_id` is the key and the latter has two keys: `course_id` and `user.id`).

Reply:
Ah, I see, that does seem to be the case. That is pretty confusing though that we had to run cache.clear() to find this out and that the code seems to imply something else. Anyway, let's just move on.